src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1729,9 +1737,19 @@ public override bool Equals([NotNullWhen(true)] object? comparand)
             MoreInfo selfInfo = _info.MoreInfo;
             MoreInfo otherInfo = obj._info.MoreInfo;
 
+            UriComponents components = UriComponents.HttpRequestUrl;
+
+            if (_syntax.InFact(UriSyntaxFlags.MailToLikeUri))
+            {
+                if (!obj._syntax.InFact(UriSyntaxFlags.MailToLikeUri))
+                    return false;
+
+                components |= UriComponents.UserInfo;
+            }
+
             // Fragment AND UserInfo are ignored
```

Comment:
```suggestion
            // Fragment AND UserInfo (for non-mailto URIs) are ignored
```
And move it to the `components` definition above